nginx/nginx-router-mesh.conf.j2

```diff
@@ -81,7 +81,7 @@ http {
 			#proxy_cache_min_uses 3;
 			proxy_cache_valid      200  30s;
 			proxy_cache_use_stale  error timeout invalid_header updating http_500 http_502 http_503 http_504;
-            proxy_pass http://{{ router_mesh.host }}/user-manager/v1/users;
+            proxy_pass http://router-mesh/user-manager/v1/users;
```

Comment:
This change means that the router-mesh service will always have the host name router-mesh. Is that intentional?

Reply:
Hey @cpretzer, in the included router-mesh config files I added router-mesh as a service so that it would be included as an upstream. What this change does is make it so that it does a proxy_pass to the router-mesh upstream, but the router-mesh upstream itself is still going to use the router_mesh.host variable. I made this change because it is more inline with how we have been arranging services as their own upstream. Configuring this proxy_pass directly to router-mesh without an upstream seemed inconsistent.

Reply:
It looks like `{{router-mesh.host}}` is the correct variable in this case.

In the change to router-mesh_config.yaml below, we see that the router mesh service is defined as:
```router_mesh:
    server: router-mesh
    scheme: http
    expose_external: true
    healthcheck: true
    healthcheck_uri: /```

and the jinja template creates services with this macro:

```{% for name, service in services.iteritems() %}
     upstream {{ name }} {	     
         server        {{ router_mesh.host }}{% if namespace != None and internal_domain_name != None %}.{{ namespace }}.{{ internal_domain_name }}{% endif %};
         zone backend  64k;	         
         keepalive 		300;	         
     }	     
}```

So, given these files, I expect the generated upstream block to be:

```
upstream router_mesh {
    server  router-mesh;
    zone backend 64k;
    keepalive 300;
}```

I don't see the `router-mesh` upstream, and I would expect NGINX to write out some errors on startup.

I'll run some tests.

Reply:
I see what you're saying. I think that "router-mesh" in this case should actually be "router_mesh" so that it properly routes through the upstream. But I think that it should route through the upstream rather than just having "router_mesh.host" because then it will work with kubernetes.